src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs

```diff
@@ -17,25 +17,47 @@ public class LogValuesFormatter
     {
         private const string NullValue = "(null)";
         private static readonly object[] EmptyArray = new object[0];
-        private static readonly char[] FormatDelimiters = {',', ':'};
-        private readonly string _format;
-        private readonly List<string> _valueNames = new List<string>();
+        private static readonly char[] FormatDelimiters = { ',', ':' };
+        private string _format;
+        private List<string> _valueNames;
 
         public LogValuesFormatter(string format)
         {
             OriginalFormat = format;
+        }
 
-            var sb = new StringBuilder();
+        public string OriginalFormat { get; private set; }
+        public List<string> ValueNames
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_format)) FormatInput(OriginalFormat); //construct the valuenames from the input
+                return _valueNames ?? new List<string>(0);
+            }
+        }
+
+        private string FormatInput(string format)
+        {
+            if (!string.IsNullOrEmpty(_format))
```

Comment:
Why isn't null check enough?

Reply:
This is initialized with an empty list because we have directly used the variable without null checks in the LoggerMessage.cs, FormattedLogValues.cs & LogFormatAnalyzer.cs. If I set null value here, those files will have to check for null and then add to the KeyValuePair. The references are listed below for your reference. Kindly confirm

![image](https://user-images.githubusercontent.com/20381669/37871832-5110b5fa-3015-11e8-9737-431bc2496d84.png)